rapids-cmake/cpm/nvcomp.cmake

```diff
@@ -0,0 +1,141 @@
+  endif()
+  if(RAPIDS_BUILD_EXPORT_SET)
+    list(APPEND RAPIDS_UNPARSED_ARGUMENTS BUILD_EXPORT_SET ${RAPIDS_BUILD_EXPORT_SET})
+  endif()
+
+  set(to_exclude OFF)
+  if(NOT RAPIDS_INSTALL_EXPORT_SET OR exclude)
+    set(to_exclude ON)
+  endif()
+
+  include("${rapids-cmake-dir}/cpm/detail/package_details.cmake")
+  rapids_cpm_package_details(nvcomp version repository tag shallow exclude)
+
+  # first see if we have a proprietary pre-built binary listed in versions.json and it if requested.
+  #
+  set(nvcomp_proprietary_binary OFF) # will be set to true by rapids_cpm_get_proprietary_binary
+  if(RAPIDS_USE_PROPRIETARY_BINARY)
+    include("${rapids-cmake-dir}/cpm/detail/get_proprietary_binary.cmake")
+    rapids_cpm_get_proprietary_binary(nvcomp ${version})
+  endif()
+
+  include("${rapids-cmake-dir}/cpm/find.cmake")
+  rapids_cpm_find(nvcomp ${version} ${RAPIDS_UNPARSED_ARGUMENTS}
+                  GLOBAL_TARGETS nvcomp::nvcomp
+                  CPM_ARGS
+                  GIT_REPOSITORY ${repository}
+                  GIT_TAG ${tag}
+                  GIT_SHALLOW ${shallow}
+                  EXCLUDE_FROM_ALL ${to_exclude}
+                  OPTIONS "BUILD_STATIC ON" "BUILD_TESTS OFF" "BUILD_BENCHMARKS OFF"
+                          "BUILD_EXAMPLES OFF")
+
+  # nvcomp creates the correct namespace aliases
+  if(NOT TARGET nvcomp::nvcomp AND TARGET nvcomp)
+    add_library(nvcomp::nvcomp ALIAS nvcomp)
+  endif()
+
+  # Propagate up variables that CPMFindPackage provide
+  set(nvcomp_SOURCE_DIR "${nvcomp_SOURCE_DIR}" PARENT_SCOPE)
+  set(nvcomp_BINARY_DIR "${nvcomp_BINARY_DIR}" PARENT_SCOPE)
+  set(nvcomp_ADDED "${nvcomp_ADDED}" PARENT_SCOPE)
+  set(nvcomp_VERSION ${version} PARENT_SCOPE)
+  set(nvcomp_proprietary_binary ${nvcomp_proprietary_binary} PARENT_SCOPE)
+
+  # Setup up install rules when using the proprietary_binary When building from source, nvcomp will
+  # set the correct install rules
+  include("${rapids-cmake-dir}/export/find_package_root.cmake")
+  if(RAPIDS_INSTALL_EXPORT_SET AND nvcomp_proprietary_binary)
+    include(GNUInstallDirs)
+    install(DIRECTORY "${nvcomp_ROOT}/lib/" DESTINATION ${CMAKE_INSTALL_LIBDIR})
+    install(DIRECTORY "${nvcomp_ROOT}/include/" DESTINATION ${CMAKE_INSTALL_INCLUDEDIR})
+    # place the license information in the location that conda uses
+    install(FILES "${nvcomp_ROOT}/NOTICE" DESTINATION info/licenses/ RENAME NVCOMP_NOTICE)
+    install(FILES "${nvcomp_ROOT}/LICENSE" DESTINATION info/licenses/ RENAME NVCOMP_LICENSE)
+  endif()
+
+  if(RAPIDS_BUILD_EXPORT_SET AND nvcomp_proprietary_binary)
+    # point our consumers to where they can find the pre-built version
+    rapids_export_find_package_root(BUILD nvcomp "${nvcomp_ROOT}" ${RAPIDS_BUILD_EXPORT_SET})
+  elseif(RAPIDS_BUILD_EXPORT_SET)
+    # point our consumers to where they can find the source version
+    rapids_export_find_package_root(BUILD nvcomp "${nvcomp_BINARY_DIR}" ${RAPIDS_BUILD_EXPORT_SET})
```

Comment:
Does `rapids_cpm_find` not handle this case?

Reply:
Droped, you are correct it isn't needed.